js/utilities.js

```diff
@@ -0,0 +1,79 @@
+import { $toaster } from "./elements.js";
+
+export const showToastMessage = (message) => {
+  $toaster.textContent = message;
+  $toaster.hidden = false;
+  setTimeout(() => {
+    $toaster.hidden = true;
+  }, 1500);
+};
+
+export const sanitizeInput = (value) => {
+  const reg = /[&<>"'/`]/gi;
+  return value.replace(reg, "").trim();
+};
+
+export const createElement = (text, type, onClick) => {
+  const $element = document.createElement(type);
+  $element.innerText = text;
+  if (onClick) {
+    $element.addEventListener("click", onClick);
+  }
+  return $element;
+};
+
+export const createTaskElement = (
+  task,
+  deleteTask,
+  editTask,
+  updateTask,
+  cancelEdit,
+  completeTask
+) => {
+  const $taskItem = document.createElement("li");
+
+  const $titleElement = document.createElement("span");
+  $titleElement.textContent = task.title;
+  if (task && task.done) {
+    $titleElement.style.textDecoration = "line-through";
+  }
+
+  if (task.isEditing) {
+    const $inputField = document.createElement("input");
+    $inputField.type = "text";
+    $inputField.value = task.title;
+
+    const $updateButton = createElement("Update", "button", () => {
+      const sanitizedTitle = sanitizeInput($inputField.value);
+      updateTask(task, sanitizedTitle);
+    });
+    const $cancelButton = createElement("Cancel", "button", () => {
+      cancelEdit(task);
+    });
+
+    $taskItem.append($inputField, $updateButton, $cancelButton);
+  } else {
+    const $deleteButton = createElement("Delete", "button", () =>
+      deleteTask(task.id)
+    );
+    const $editButton = createElement("Edit", "button", () => editTask(task));
+    const $doneButton = createElement("Done", "button", () =>
+      completeTask(task.id)
+    );
```

Comment:
For better readability create the buttons like this - `const deleteButton = document.createElement("button");` and then you can use addEventListeners accordingly.

Reply:
Here I've created a modular function to handle element creation. So I am passing the element name, element type and the handler function. Then the modular function creates that element for me which helps us to obtain modularity in this project.